src/libstd/process.rs

```diff
@@ -1418,7 +1418,8 @@ mod tests {
 
         for (ref k, ref v) in env::vars() {
             // don't check android RANDOM variables
-            if cfg!(target_os = "android") && *k == "RANDOM" {
+            if cfg!(target_os = "android") &&
+               (*k == "RANDOM" || k.contains("-")) {
```

Comment:
FWIW environment variables with `-` is pretty rare (most have `_`) and I'm just sort of randomly chalking this up to a bug in the `set` command which we're using to learn about environment variables.